### PR TITLE
doc: added documentation to API methods

### PR DESCRIPTION
**Pull request by Lucca-C**
Wrote JSdocs for relevant methods.

### CODE REVIEW

File: js/shipment.js [hunk 1]
```diff
@@ -104,6 +120,14 @@ export const ShipmentAPI = superclass =>
             return response;
         }
 
+        /**
+         *
+         * Gets proof of delivery of shipment.
+         *
+         * @param {String} trackingNumber is the number of the shipment/waybill.
```
Comment (joao-conde) [inline js/shipment.js:127]: Fix like above.

**Review by joao-conde — CHANGES_REQUESTED** [review]

**Review by joao-conde — COMMENTED** [review]
@Lucca-C besides fixing the issues below, make sure to look at:
https://github.com/ripe-tech/ups-api-js/pull/28/files

(my PR to the UPS API Client that I showed you)

See the changes I made in the README and try to replicate in this PR for DHL.

File: README.md [hunk 2]
```diff
@@ -2,6 +2,20 @@
 
 The DHL API Javascript client implementation.
 
+```javascript
+const dhlAPI = require("dhl-api");
+
+// instance the DHL client with defaults
```
Comment (joao-conde) [inline README.md:8]: better "// instance the API client with defaults" to keep parity with the UPS one

File: README.md [hunk 3]
```diff
@@ -2,6 +2,20 @@
...
+const dhl = new dhlApi.API({
+    username: "mydhlaccount",
+    password: "mydhlaccountpassword",
+    license: "3F9955C3F789C255"
+});
```
Comment (joao-conde) [inline README.md:13]: These parameters are wrong. Take a look at the mandatory ones (the ones with no default for [UPS  here](https://github.com/ripe-tech/ups-api-js/blob/master/js/base.js#L31-L33))

Now check the [constructor for the DHL one](https://github.com/ripe-tech/dhl-api-js/blob/master/js/base.js#L10). Notice we can give username password or the token itself (which defaults to user + password).

I would therefore document username and password only.

File: README.md [hunk 4]
```diff
@@ -2,6 +2,20 @@
...
+
+// example: request tracking information
+const tracking = await api.getTrackingDetails("7798339175");
```
Comment (joao-conde) [inline README.md:16]: Does this method exist in the DHL package

File: js/base.js [hunk 5]
```diff
@@ -45,6 +45,7 @@ export class API extends mix(BaseAPI).with(PickupAPI, ShipmentAPI, TrackingAPI)
      * made by the API client
      * @returns {Object|String|Blob} The parsed result value for the provided
      * response object taking into account the content type of it.
+     * @see https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section/operations-pickup-exp-api-pickups
```
Comment (joao-conde) [inline js/base.js:48]: Notice that this does not make sense. This function of ours is not documented in that link (DHL API documentation)

File: js/pickup.js [hunk 6]
```diff
@@ -1,5 +1,13 @@
 export const PickupAPI = superclass =>
     class extends superclass {
+        /**
+         * Schedules a pickup for a parcel.
+         *
+         * @param {Object} payload The payload object according to the DHL API standards.
+         * @param {Object} options An object of options to configure the request.
+         * @returns {Object} The HTTP response object.
+         * @see https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section/operations-pickup-exp-api-pickups
```
Comment (joao-conde) [inline js/pickup.js:9]: That link takes you to the same place as just `"https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section"`. So let's use this one

File: js/shipment.js [hunk 7]
```diff
@@ -86,6 +86,14 @@ export const GO_GREEN_LOGO =
 
 export const ShipmentAPI = superclass =>
     class extends superclass {
+        /**
+         * Creates a shipment and associated DHL waybill.
+         *
+         * @param {Object} payload The payload object according to the DHL API standards.
+         * @param {Object} options An object of options to configure the request.
+         * @returns {Object} The HTTP response object.
+         * @see https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section/operations-pickup-exp-api-pickups
```
Comment (joao-conde) [inline js/shipment.js:95]: Notice that you copy-pasted the pickup one and this is the shipment method. So at the very least would be a different URL. Regardless, let's use "https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section" for the same reason as before.

File: js/shipment.js [hunk 8]
```diff
@@ -95,6 +103,15 @@ export const ShipmentAPI = superclass =>
...
         }
 
+        /**
+         * Uploads a document to the DHL server for a shipment.
+         *
+         * @param {String} trackingNumber The tracking number of the shipment/waybill
+         * @param {Object} payload The payload object according to the DHL API standards.
+         * @param {Object} options An object of options to configure the request.
+         * @returns {Object} The HTTP response object.
+         * @see https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section/operations-pickup-exp-api-pickups
```
Comment (joao-conde) [inline js/shipment.js:113]: Let's use "https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section". Do this in the rest of the links

File: js/shipment.js [hunk 9]
```diff
@@ -104,6 +120,14 @@ export const ShipmentAPI = superclass =>
             return response;
         }
 
+        /**
+         *
```
[the remark this answers is not in the document]
Comment (joao-conde) [reply js/shipment.js:125]: Still not solved

File: js/tracking.js [hunk 10]
```diff
@@ -74,6 +74,14 @@ export const TrackingAPI = superclass =>
             return `https://www.dhl.com/global-en/home/tracking/tracking-express.html?submit=1&tracking-id=${trackingNumber}`;
         }
 
+        /**
+         *
```
[the remark this answers is not in the document]
Comment (joao-conde) [reply js/tracking.js:78]: Still not solved

**Review by joao-conde — CHANGES_REQUESTED** [review]

**Comment by joao-conde** [comment]
@Lucca-C besides the requested changes, also update the description on your Pull Request please :)

File: js/tracking.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (Lucca-C) [reply js/tracking.js:78]: I think the linter is adding the new lines.

**Review by joao-conde — APPROVED** [review]

File: js/pickup.js [hunk 11]
```diff
@@ -1,5 +1,13 @@
 export const PickupAPI = superclass =>
     class extends superclass {
+        /**
+         * Schedules a pickup for a parcel.
+         *
+         * @param {Object} payload The payload object according to the DHL API standards.
+         * @param {Object} options An object of options to configure the request.
+         * @returns {Object} The HTTP response object.
+         * @see https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section
```
Comment (BeeMargarida) [inline js/pickup.js:9]: Shouldn't this be:
```suggestion
         * @see https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section/operations-tag-pickup
```

Specific to the endpoint you are using here

File: js/shipment.js [hunk 12]
```diff
@@ -86,6 +86,14 @@ export const GO_GREEN_LOGO =
 
 export const ShipmentAPI = superclass =>
     class extends superclass {
+        /**
+         * Creates a shipment and associated DHL waybill.
+         *
+         * @param {Object} payload The payload object according to the DHL API standards.
+         * @param {Object} options An object of options to configure the request.
+         * @returns {Object} The HTTP response object.
+         * @see https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section
```
Comment (BeeMargarida) [inline js/shipment.js:95]:
```suggestion
         * @see https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section/operations-tag-shipment

```

File: js/shipment.js [hunk 13]
```diff
@@ -95,6 +103,15 @@ export const ShipmentAPI = superclass =>
...
         }
 
+        /**
+         * Uploads a document to the DHL server for a shipment.
+         *
+         * @param {String} trackingNumber The tracking number of the shipment/waybill.
+         * @param {Object} payload The payload object according to the DHL API standards.
+         * @param {Object} options An object of options to configure the request.
+         * @returns {Object} The HTTP response object.
+         * @see https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section
```
Comment (BeeMargarida) [inline js/shipment.js:113]:
```suggestion
         * @see https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section/operations-tag-shipment

```

File: js/shipment.js [hunk 14]
```diff
@@ -95,6 +103,15 @@ export const ShipmentAPI = superclass =>
             return response;
         }
 
+        /**
+         * Uploads a document to the DHL server for a shipment.
```
Comment (BeeMargarida) [inline js/shipment.js:107]:
```suggestion
         * Uploads a document for a shipment to the DHL server.

```

File: js/shipment.js [hunk 15]
```diff
@@ -104,6 +121,14 @@ export const ShipmentAPI = superclass =>
             return response;
         }
 
+        /**
+         * Gets proof of delivery document for a shipment.
+         *
+         * @param {String} trackingNumber The tracking number of the shipment/waybill.
+         * @param {Object} options An object of options to configure the request.
+         * @returns {Object} The HTTP response object.
+         * @see https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section
```
Comment (BeeMargarida) [inline js/shipment.js:130]:
```suggestion
         * @see ﻿https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section/operations-tag-shipment

```

File: js/shipment.js [hunk 16]
```diff
@@ -104,6 +121,14 @@ export const ShipmentAPI = superclass =>
             return response;
         }
 
+        /**
+         * Gets proof of delivery document for a shipment.
```
Comment (BeeMargarida) [inline js/shipment.js:125]:
```suggestion
         * Gets proof of delivery document of a shipment.

```

File: js/tracking.js [hunk 17]
```diff
@@ -74,6 +74,14 @@ export const TrackingAPI = superclass =>
             return `https://www.dhl.com/global-en/home/tracking/tracking-express.html?submit=1&tracking-id=${trackingNumber}`;
         }
 
+        /**
+         * Gets shipment tracking information for a shipment.
```
Comment (BeeMargarida) [inline js/tracking.js:78]:
```suggestion
         * Gets the tracking information of a shipment

```

File: js/tracking.js [hunk 18]
```diff
@@ -74,6 +74,14 @@ export const TrackingAPI = superclass =>
             return `https://www.dhl.com/global-en/home/tracking/tracking-express.html?submit=1&tracking-id=${trackingNumber}`;
         }
 
+        /**
+         * Gets shipment tracking information for a shipment.
+         *
+         * @param {String} trackingNumber The tracking number of the shipment/waybill.
+         * @param {Object} options An object of options to configure the request.
+         * @returns {Object} The HTTP response object.
+         * @see https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section
```
Comment (BeeMargarida) [inline js/tracking.js:83]:
```suggestion
         * @see https://developer.dhl.com/api-reference/dhl-express-mydhl-api#reference-docs-section/operations-tag-tracking

```

**Review by BeeMargarida — CHANGES_REQUESTED** [review]

File: js/pickup.js (hunk 11 above)
Comment (Lucca-C) [reply js/pickup.js:9]: I believe those links point to the same place, happy to change them if you think it's clearer.

Comment (BeeMargarida) [reply js/pickup.js:9]: Hum, taking that into consideration, there is no need to change this 👍

**Review by BeeMargarida — APPROVED** [review]

File: js/shipment.js [hunk 19]
```diff
@@ -95,6 +103,16 @@ export const ShipmentAPI = superclass =>
             return response;
         }
 
+        /**
+         * Uploads a document for a shipment to the DHL server.
+
```
Comment (joamag) [inline js/shipment.js:108]: remove extra EOL

File: js/tracking.js [hunk 20]
```diff
@@ -74,6 +74,15 @@ export const TrackingAPI = superclass =>
             return `https://www.dhl.com/global-en/home/tracking/tracking-express.html?submit=1&tracking-id=${trackingNumber}`;
         }
 
+        /**
+         * Gets the tracking information of a shipment
+
```
Comment (joamag) [inline js/tracking.js:79]: remove extra EOL

File: js/tracking.js [hunk 21]
```diff
@@ -74,6 +74,15 @@ export const TrackingAPI = superclass =>
             return `https://www.dhl.com/global-en/home/tracking/tracking-express.html?submit=1&tracking-id=${trackingNumber}`;
         }
 
+        /**
+         * Gets the tracking information of a shipment
```
Comment (joamag) [inline js/tracking.js:78]: add `.`

**Review by joamag — CHANGES_REQUESTED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**